spec/commands_generator_spec.rb

```diff
@@ -0,0 +1,9 @@
+require "gym/commands_generator"
+
+describe Gym do
+  describe Gym::CommandsGenerator do
+    it "validates available_options" do
+      # this runs the initialization code in the commands generator and detects invalid options
```

Comment:
I added the test because I had added an option (for the dry_run option) that actually caused a problem (I hadn't specified a short_option, and th CommandsGenerator does trigger the adding of a default short_option based on the option name, which caused a conflict with a pre-existing option).

But I agree that this relies too much on code executing on its own...

Another way would be to run the --help option as post build check of all the commands and check the return value.

Better idea ?